### PR TITLE
Rudimentary validator for binary files

### PR DESCRIPTION
**Pull request by XapaJIaMnu**
Validate that the binary file you are trying to load is not truncated.

### CODE REVIEW

File: src/translator/byte_array_util.cpp [hunk 1]
```diff
@@ -5,6 +5,75 @@
 namespace marian {
 namespace bergamot {
 
+// This is a basic validator that checks if the file has not been truncated
+// it basically loads up the header and checks
+
+// This struct and the getter are copied from the marian source, because it's located
+// inside src/common/binary.cpp:15 and we can't include it.
```
Comment (jerinphilip) [inline src/translator/byte_array_util.cpp:12]: /// to be picked up by doxygen, consistent with what exists in source.

Comment (XapaJIaMnu) [reply src/translator/byte_array_util.cpp:12]: I don't want this to be picked up by doxygen? It's an internal function, not exposed by the headers and only really meant to be read by the developers.

File: src/translator/byte_array_util.cpp [hunk 2]
```diff
@@ -5,6 +5,75 @@
...
+template <typename T>
+const T* get(const void*& current, uint64_t num = 1) {
+  const T* ptr = (const T*)current;
+  current = (const T*)current + num;
+  return ptr;
+}
+
+bool validateBinaryModel(AlignedMemory& model, uint64_t fileSize) {
+  const void * current = &model[0];
+  uint64_t memoryNeeded = sizeof(uint64_t)*2; // We keep track of how much memory we would need if we have a complete file
```
Comment (jerinphilip) [inline src/translator/byte_array_util.cpp:30]: wrap lines with comments to 80 characters is standard? My config overwrites this autoformat and I always end up bearing the brunt.

Comment (XapaJIaMnu) [reply src/translator/byte_array_util.cpp:30]: Dude, we're not 1998, we have 4k monitors. Just. No. All modern projects (including marian) have abandoned that.

File: src/translator/byte_array_util.cpp (hunk 1 above)
Comment (kpu) [reply src/translator/byte_array_util.cpp:12]: Anonymous namespace please.  

The standard for documentation is higher in this repo than it is elsewhere, sorry @XapaJIaMnu.

File: src/translator/byte_array_util.cpp (hunk 2 above)
Comment (kpu) [reply src/translator/byte_array_util.cpp:30]: I'm ok with longer lines.

File: src/translator/byte_array_util.cpp (hunk 1 above)
Comment (XapaJIaMnu) [reply src/translator/byte_array_util.cpp:12]: Sorry, I genuinely do not understand how I should document this better. I'm went through the repo and looked at various files: (this one comes to mind) https://github.com/browsermt/bergamot-translator/blob/main/src/translator/request.h

Do I need to add the `defines` bit on top?

Comment (jerinphilip) [reply src/translator/byte_array_util.cpp:14]: I think what @kpu means is to either leave it in an anonymous namespace if it's local and don't bother documenting, or fit it with doxygen compatible documentation.

[This](https://github.com/browsermt/bergamot-translator/blob/jp/response-builder/src/translator/request.h) is the request which is documented. The target is a sphinx doc build, a demo of which looks like [this](https://jerinphilip.github.io/bergamot-translator/api/namespace_marian__bergamot.html).

Comment (XapaJIaMnu) [reply src/translator/byte_array_util.cpp:14]: Ok that should be fixed now.

File: src/translator/byte_array_util.cpp [hunk 3]
```diff
@@ -5,6 +5,79 @@
...
+
+  // If this final check passes, the file is at least big enough to contain the model
+  if (fileSize >= memoryNeeded) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+} // Annonymous namespace
```
Comment (kpu) [inline src/translator/byte_array_util.cpp:79]: } // namespace

**Review by kpu — APPROVED** [review]

**Review by abhi-agg — APPROVED** [review]
Approving this change (as Kenneth already approved + CI is passing + All the unresolved conversations seem to be resolved) so that we can merge it 👍

**Pull request merged**